students/mattmaeda/session07/html_render.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python3
+# coding=utf-8
+from __future__ import unicode_literals
+
+class Element(object):
+    """ class for rendering HTML element
+    """
+    def __init__(self, content=None, **kwargs):
+        self.tag = "tag"
+        self.elements = []
+        self.kwargs = {}
+
+        if content is not None:
+            self.append(content)
+
+        if kwargs is not None:
+            self.kwargs = kwargs
+
+    def append(self, content):
+        """ Appends content to class attribute elements list
+        """
+        self.elements.append(content)
+
+
+    def render(self, file_out, ind="", depth=1):
+        """ Writes content to StringIO buffer with optional indentation
+        :file_out StringIO: StringIO
+        :ind String: char indentation
+        """
+        formatted_open_tag = self.tag
+        for key, value in self.kwargs.items():
+            formatted_open_tag += ' {}="{}"'.format(key, value)
+
+        file_out.write("{}<{}>\n".format(ind*depth, formatted_open_tag))
+
+        for elem in self.elements:
+            if isinstance(elem, str):
+                file_out.write("{}{}\n".format(ind*(depth+1), elem))
+            else:
+                elem.render(file_out, ind, depth+1)
+
+        else:
+            file_out.write("{}</{}>\n".format(ind*depth, self.tag))
+
+
+class OneLineTag(Element):
+    def __init__(self, content=None, **kwargs):
+        super(OneLineTag, self).__init__(content, **kwargs)
+
+
+    def render(self, file_out, ind = "", depth=1):
+        for elem in self.elements:
```

Comment:
this is assuming there will only be text in this element -- probably be a safe assumption of a one-line tag -- but maybe actually enforce that there is only one piece of content if you're going to do that.